analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/ConstantCheck.cs

```diff
@@ -50,11 +50,12 @@ public static SymbolicConstraint ConstraintFromType(ITypeSymbol type)
             operation.Instance.ConstantValue.Value switch
             {
                 // Update DefaultValue when adding new types
-                true => SymbolicValue.True,
-                false => SymbolicValue.False,
+                true => SymbolicValue.True.WithConstraint(ObjectConstraint.NotNull),
+                false => SymbolicValue.False.WithConstraint(ObjectConstraint.NotNull),
                 null when (operation.Instance.Type ?? ConvertedType(operation.Parent)).CanBeNull() => SymbolicValue.Null,  // ToDo: MMF-2401, this will need to be reviewed
                 string => SymbolicValue.NotNull,
-                _ => null
+                ValueType => SymbolicValue.NotNull,
```

Comment:
`Nullable<T>` also derives from `ValueType` but we don't care here. Constant expressions are never of type `Nullable<something>` [according to the spec](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#1121-constant-expressions) and the `null` constant case is handled before.